README.md

```diff
@@ -4,42 +4,59 @@ Provides Azure Authentication Middleware for a Laravel App.  If you like this, c
-8. Within our app on https://portal.azure.com/ point `reply url` to the `/login/azurecallback` route with the full url (ex: http://thewebsite.com/login/azurecallback).
-9. Add the `azure` middleware to your route groups on any routes that needs protected by auth and enjoy :tada:
+4. Open `App/Http/Kernel.php` and add `'azure' => \RootInc\LaravelAzureMiddleware\Azure::class,` to the `$routeMiddleware` array to register the middleware.
+5. Open `.env` and add the following variables (with your own values):
+```
+AZURE_TENANT_ID=your-azuread-tenant-id
+AZURE_CLIENT_ID=your-app-registration-application-id
+AZURE_CLIENT_SECRET=your-app-registration-client-secret
+AZURE_RESOURCE=https://graph.microsoft.com
+```
+All these values can be gotten from your tenant through https://portal.azure.com.
+6. As of version 0.8.0, the variable `AZURE_SCOPE` was added to the project, which are permissions to be used for the request.  You can read more about these here: https://docs.microsoft.com/en-us/graph/api/resources/users?view=graph-rest-1.0
+7. Additionally, an optional variable, `AZURE_DOMAIN_HINT` was added, so it can be used to help users know which email address they should login with.  More info here: https://azure.microsoft.com/en-us/updates/app-service-auth-and-azure-ad-domain-hints/
+8. In your App Registration on https://portal.azure.com/, set the `Redirect URIs` (often referred to as reply URLs) to the `/azure/callback` route with the full url (example: https://yourwebsite.com/azure/callback or http://localhost:8000 when running development locally).
+9. Add the `azure` middleware to your route groups on any routes that needs to be protected by authentication and enjoy :tada:
 10. If you need custom callbacks, see [Extended Installation](#extended-installation).
 
-__NOTE: ~~You may need to add premissions for (legacy) Azure Active Directory Graph~~ As of 0.8.0, we are using v2 of Azure's login API, which allows us to pass scopes, or permissions we'd like to use.__
+__NOTE: As of version 0.8.0, the project uses v2 of Azure's login API, which allows it to pass scopes, or permissions that can used.__
 
 ## Routing
 
-`Route::get('/login/azure', '\RootInc\LaravelAzureMiddleware\Azure@azure');` First parameter can be wherever you want to route the azure login.  Change as you would like.
-
-`Route::get('/login/azurecallback', '\RootInc\LaravelAzureMiddleware\Azure@azurecallback');` First parameter can be whatever you want to route after your callback.  Change as you would like.
+As shown in [Normal Installation](#normal-installation), step 3.
+The route path and name can be whatever you need it to be.
 
-`Route::get('/logout/azure', '\RootInc\LaravelAzureMiddleware\Azure@azurelogout');` First parameter can be whatever you want to route after your callback.  Change as you would like.
+For example, instead of 
+```php
+Route::get('/azure/login', [\RootInc\LaravelAzureMiddleware\Azure::class,'azure'])->name('azure.login');
```

Comment:
Is the array binding faster?  I tend to like the readability of the other better.

Reply:
This make code hinting work for functions inside the controller and is defined in the [Laravel 8 documentation](https://laravel.com/docs/8.x/routing#the-default-route-files), thus I consider it to be best practice. Thoughts?

Reply:
Interesting.  Back in my day (Laravel 5.4), it was written the as a string.  If Laravel 8's docs are written as such, let's proceed with the suggested change.